pkgs/development/tools/build-managers/gradle/default.nix

```diff
@@ -78,18 +78,36 @@ rec {
 
       fixupPhase = let arch = if stdenv.is64bit then "amd64" else "i386";
       in ''
+        nativePlatformJar=$(basename $(ls -1 $out/lib/gradle/lib/native-platform-*.jar | grep -Ev '.-(linux|osx|freebsd|windows)-'))
+        nativeVersion=''${nativePlatformJar#native-platform-}
+        nativeVersion=''${nativeVersion%.jar}
         for variant in "" "-ncurses5" "-ncurses6"; do
-          mkdir "patching$variant"
-          pushd "patching$variant"
-          jar xf $out/lib/gradle/lib/native-platform-linux-${arch}$variant-${nativeVersion}.jar
+          mkdir "patching-native-platform$variant"
+          pushd "patching-native-platform$variant"
+          jar xf $out/lib/gradle/lib/native-platform-linux-${arch}$variant-''${nativeVersion}.jar
           patchelf \
             --set-rpath "${stdenv.cc.cc.lib}/lib64:${lib.makeLibraryPath [ stdenv.cc.cc ncurses5 ncurses6 ]}" \
             net/rubygrapefruit/platform/linux-${arch}$variant/libnative-platform*.so
-          jar cf native-platform-linux-${arch}$variant-${nativeVersion}.jar .
-          mv native-platform-linux-${arch}$variant-${nativeVersion}.jar $out/lib/gradle/lib/
+          jar cf native-platform-linux-${arch}$variant-''${nativeVersion}.jar .
+          mv native-platform-linux-${arch}$variant-''${nativeVersion}.jar $out/lib/gradle/lib/
           popd
         done
 
+        # The file-events library _seems_ to follow the native-platform version, but
+        # we won’t assume that.
+        fileEventsJar=$(basename $(ls -1 $out/lib/gradle/lib/file-events-*.jar | grep -Ev '.-(linux|osx|freebsd|windows)-'))
```

Comment:
Maybe define a function to get the version like this? It seems to be the same code in lines 81-83 above.

Reply:
Added a helper and switched to using autoPatchelf, since it does exactly what we need here.